File: javascript/word-count/word-count.js
class Words {
  count(words) {
    return words
      .toLowerCase()
      .trim()
      .split(/\s/)
      .reduce((map, word) => {
        if (word === '') return map;
        if (map[word] && typeof map[word] === 'number') map[word]++;
        else map[word] = 1;
        return map;
      }, {});
  }
}

export default Words;

File: swift/word-count/Sources/WordCount/WordCount.swift
import Foundation

struct WordCount {
    let words: String
    init(words: String) {
        self.words = words.lowercased().replacingOccurrences(
            of: #"[^0-9a-z ]"#,
            with: "",
            options: .regularExpression
        )
    }
    
    func count() -> [String: Int] {
        var res = [String: Int]()
        
        for word in words.split(separator: " ") {
            if res.keys.contains(String(word)) {
                res[String(word)]! += 1
            } else {
                res[String(word)] = 1
            }
        }
        return res
    }
}
